tests: net: socket: tls move to new ztest API

hunterhu3000 opened the pull request:
Move net/socket/tls tests to use new ztest API
TEST=twister -T tests/net/socket/tls

Signed-off-by: Hu Zhenyu <zhenyu.hu@intel.com>

enjiamai commented on tests/net/socket/tls/src/main.c:
```diff
@@ -611,26 +611,4 @@ void test_v6_dtls_sendmsg(void)
-				K_PRIO_COOP(CONFIG_NUM_COOP_PRIORITIES - 1));
-	} else {
-		k_thread_priority_set(k_current_get(), K_PRIO_PREEMPT(8));
-	}
```
Should Line 614~621 be added to the setup function?

hunterhu3000 committed:
tests: net: socket: tls move to new ztest API

Move net/socket/tls tests to use new ztest API
TEST=twister -T tests/net/socket/tls

Signed-off-by: Hu Zhenyu <zhenyu.hu@intel.com>

hunterhu3000 commented:
This test case cannot be run with CONFIG_ZTEST_SHUFFLE =y, as some of the test case cannot be run twice, eg test_v6_msg_trunc
Thanks to review

enjiamai commented:
Re-review it, good for me.
